addons/im_livechat/static/tests/new/unread_messages_tests.js

```diff
@@ -0,0 +1,69 @@
+/** @odoo-module **/
+
+import { startServer } from "@bus/../tests/helpers/mock_python_environment";
+
+import {
+    setCookie,
+    start,
+    loadDefaultConfig,
+    waitUntil,
+} from "@im_livechat/../tests/helpers/new/test_utils";
+
+import { Command } from "@mail/../tests/helpers/command";
+import { afterNextRender } from "@mail/../tests/helpers/test_utils";
+
+QUnit.module("thread service");
+
+QUnit.test("unread counter", async (assert) => {
+    const pyEnv = await startServer();
+    const livechatChannelId = await loadDefaultConfig();
+    const channelId = pyEnv["discuss.channel"].create({
+        channel_member_ids: [
+            Command.create({ partner_id: pyEnv.currentPartnerId }),
+            Command.create({ partner_id: pyEnv.publicPartnerId }),
+        ],
+        channel_type: "livechat",
+        livechat_channel_id: livechatChannelId,
+        livechat_operator_id: pyEnv.currentPartnerId,
+    });
+    const [channelInfo] = pyEnv.mockServer._mockDiscussChannelChannelInfo([channelId]);
+    setCookie("im_livechat_session", JSON.stringify(channelInfo));
+    const { env, root } = await start();
+    await afterNextRender(() => {
+        env.services.rpc("/mail/message/post", {
+            context: { mockedUserId: pyEnv.currentUserId },
+            post_data: { body: "Are you there?", message_type: "comment" },
+            thread_id: channelId,
+            thread_model: "discuss.channel",
+        });
+    });
+    assert.containsOnce(root, ".o-mail-ChatWindow-header:contains((1))");
+});
```

Comment:
I prefer naming test in a way that allow us to understand it without actually reading the test.
Especially when the test is just a single assertion. The name of test should match this single assertion.
Here I think it should be something like "New message from operator should display unread counter"